From 0eeaffc50948576a99c0bbc9880e65cd198955cd Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Sun, 25 Dec 2016 14:30:52 +0200
Subject: [PATCH 1255/2241] fix: axim: fix overflow in transaction calculation

Missing cast to u64 caused transaction/sec calculation to
overflow

Change-Id: I73b26db34e451aba7887c3dcbd51054e19019564
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34962
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/hwtracing/coresight/coresight-axi-monitor.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/hwtracing/coresight/coresight-axi-monitor.c b/drivers/hwtracing/coresight/coresight-axi-monitor.c
index 6f0928a..ada295b 100644
--- a/drivers/hwtracing/coresight/coresight-axi-monitor.c
+++ b/drivers/hwtracing/coresight/coresight-axi-monitor.c
@@ -433,7 +433,8 @@ static ssize_t prof_counters_show(struct device *dev,
 
 	/* # of transactions */
 	trans = readl(axim->base + AXI_MON_PR_SMP_TRANS);
-	size += scnprintf(buf + size, PAGE_SIZE, "Trans   - %10u [%d trans/sec]\n", trans, trans * 1000 / msec);
+	size += scnprintf(buf + size, PAGE_SIZE, "Trans   - %10u [%d trans/sec]\n", trans,
+			  (u32)((u64)trans * 1000 / msec));
 
 	/* # of AXI beats */
 	val = readl(axim->base + AXI_MON_PR_SMP_BEATS);
-- 
2.7.4

